### PR TITLE
Try to use array literals on reflection plugin

### PR DESCRIPTION
**Pull request by soywiz**
This PR will try to fix https://github.com/jtransc/jtransc/issues/206

### CODE REVIEW

File: jtransc-rt/resources/cpp/Base.cpp [hunk 1]
```diff
@@ -495,18 +523,45 @@ struct JA_I : JA_Base<int32_t> {
 	static JA_I *fromArgValues(int32_t a0, int32_t a1, int32_t a2) { return (JA_I * )(new JA_I(3))->init(0, a0)->init(1, a1)->init(2, a2); };
 	static JA_I *fromArgValues(int32_t a0, int32_t a1, int32_t a2, int32_t a3) { return (JA_I * )(new JA_I(4))->init(0, a0)->init(1, a1)->init(2, a2)->init(3, a3); };
 
+	static JA_I* fromArray(std::wstring desc, std::vector<int32_t> array) {
```
Comment (soywiz) [inline jtransc-rt/resources/cpp/Base.cpp]: @intrigus do you know how can I make generic all those methods with templates? so I can avoid creating duplicating them?

**Comment by soywiz** [comment]
@SergeyLabutin it just implements the first trick for reducing complexity of the function. But still lacks splitting in several functions. I will do that part tomorrow. Now I'm going to sleep. Check if this fixes your specific problem so you can continue. If not, I will try tomorrow to do the splitting part.

**Comment by soywiz** [comment]
Ok. No problem. I will complete it later today

File: jtransc-gen-cpp/src/com/jtransc/gen/cpp/CppTarget.kt [hunk 2]
```diff
@@ -966,7 +966,7 @@ class CppGenerator(injector: Injector) : CommonGenerator(injector) {
 		}
 	}
 
-	override fun genExprStringArrayLit(e: AstExpr.STRINGARRAY_LITERAL): String {
+	override fun genExprObjectArrayLit(e: AstExpr.OBJECTARRAY_LITERAL): String {
 		noImpl("C++ genExprStringArrayLit")
```
Comment (intrigus) [inline jtransc-gen-cpp/src/com/jtransc/gen/cpp/CppTarget.kt]: Should be `noImpl("C++ genExprObjectArrayLit")

File: jtransc-rt/resources/hx/JA_L.hx [hunk 3]
```diff
@@ -15,13 +15,20 @@ class JA_L extends JA_0 {
     }
 
 	{{ HAXE_METHOD_ANNOTATIONS }}
-    static public function fromArray(items:Array<Dynamic>, desc:String):JA_L {
+    static public function fromArray(desc:String, items:Array<Dynamic>):JA_L {
```
Comment (intrigus) [inline jtransc-rt/resources/hx/JA_L.hx]: Why did you switch the parameters?
This can always lead to bugs that are hard to debug.

Comment (soywiz) [inline jtransc-rt/resources/hx/JA_L.hx]: After some thinking I found that I wanted to be consisent with Tx methods. I prefer to keep variadic part to the end. Thats why I changed the order.
My original thinking was that desc is less relevant and could be even optional so i put it after desc and was an addition to analog methods in other primitive array types. But in the end, an object array need it to be correct.
Also probably i will have to change that string into a proper Type object later to be able to implement instanceof correctly for arrays without tol much overhead. I have to think this good. But probably signature will change again before 1.0.

File: jtransc-rt/resources/cpp/Base.cpp (hunk 1 above)
Comment (intrigus) [inline jtransc-rt/resources/cpp/Base.cpp]: I'm on vacation till Friday. 
But I don't know cpp templates well, so I don't know a solution :(

**Comment by soywiz** [comment]
@SergeyLabutin https://github.com/jtransc/jtransc/pull/207/commits/3a2554d5c0e87167c6e021ac4905d027dc132435 I'm doing the split here. Please check.

**Comment by SergeyLabutin** [comment]
Works fine. Thanks!

**Pull request merged**